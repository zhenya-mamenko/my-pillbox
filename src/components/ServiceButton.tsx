import { ComponentProps, } from 'react';
import { Pressable, StyleSheet, View, } from 'react-native';
import MaterialIcons from '@expo/vector-icons/MaterialIcons';

import useThemeColor from '@/hooks/useThemeColor';


interface Props extends ComponentProps<any> {
  icon: any;
  onPress: () => void;
}

export default function CircleButton({ icon, onPress }: Props) {

  const styles = StyleSheet.create({
    circleButtonContainer: {
      borderColor: useThemeColor('borderDim'),
      borderRadius: 21,
      borderWidth: 1,
      height: 42,
      margin: 5,
      padding: 3,
      width: 42,
    },
    circleButton: {
      alignItems: 'center',
      backgroundColor: useThemeColor('background'),
      borderRadius: 21,
      flex: 1,
      justifyContent: 'center',
    },
  });

  return (
    <View style={styles.circleButtonContainer}>
      <Pressable style={styles.circleButton} onPress={onPress}>
        <MaterialIcons
          color={ useThemeColor('secondary') }
          name={icon}
          size={18}
        />
      </Pressable>
    </View>
  );
}
