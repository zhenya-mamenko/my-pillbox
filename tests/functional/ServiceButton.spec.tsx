import { userEvent, render, screen, waitFor } from '@testing-library/react-native';
import ServiceButton from '@/components/ServiceButton';


jest.mock('@/hooks/useThemeColor', () => (color: string) => {
  const colors: any = {
    background: '#ffffff',
    borderDim: '#ccccccc',
    secondary: '#555555',
  }
  return colors[color];
});

jest.mock('@expo/vector-icons/MaterialIcons', () => 'MaterialIcons');

describe('ServiceButton', () => {

  it('Should render', () => {
    render(<ServiceButton icon="edit" onPress={() => {}} />);
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('Should call onPress', async () => {
    const onPress = jest.fn();
    render(<ServiceButton icon="edit" onPress={onPress} />);
    await waitFor(async () => {
      expect(await screen.getByTestId('serviceButton-edit')).toBeTruthy();
    });
    const user = userEvent.setup();
    await user.press(screen.getByTestId('serviceButton-edit'));
    expect(onPress).toHaveBeenCalled();
  });

});
