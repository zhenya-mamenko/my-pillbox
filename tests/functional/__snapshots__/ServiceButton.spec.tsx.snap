// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceButton Should render 1`] = `
<View
  style={
    {
      "borderColor": "#ccccccc",
      "borderRadius": 21,
      "borderWidth": 1,
      "height": 42,
      "margin": 5,
      "padding": 3,
      "width": 42,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderRadius": 21,
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID="serviceButton-edit"
  >
    <MaterialIcons
      color="#555555"
      name="edit"
      size={18}
    />
  </View>
</View>
`;
